server/metadata/tenant.go

```diff
@@ -1051,6 +1051,33 @@ func (tenant *Tenant) ListProjects(_ context.Context) []string {
 	return projects
 }
 
+// DeleteTenant is used to delete tenant and all the content within it.
+// This needs to be followed up with a "restart" of server to clear memory state.
+// Be careful calling this.
+func (m *TenantManager) DeleteTenant(ctx context.Context, tx transaction.Tx, tenantToDelete *Tenant) error {
+	tenantToDelete.RLock()
+	defer tenantToDelete.RUnlock()
+	projects := tenantToDelete.ListProjects(ctx)
+
+	deletedProjects := true
+	for _, project := range projects {
+		_, err := tenantToDelete.DeleteProject(ctx, tx, project)
+		if err != nil {
+			log.Error().Err(err).Msgf("Failed to delete project named %s", project)
+			deletedProjects = false
+			err = tx.Rollback(ctx)
```

Comment:
Do we need to roll back here or just return an error and let the caller roll back as caller is responsible for starting the transaction?

Reply:
you are right. This was copied code when this logic was placed within the caller. fixed it.